components/common/EventPopup/AcmEventHeads.tsx

```diff
@@ -0,0 +1,130 @@
+    const [searchResult, setSearchResult] = useState<Pill[]>([])
+    const [showSearchResults, setShowSearchResults] = useState(true)
+    const { data: session } = useSession();
+    const token = getCookieData(session).data.token
+
+    const searchUsersApi = useCallback((pattern: string) => {
+        getMatchingUsersApi(pattern, token)
+            .then((response: User[]) => {
+                // take first 5 only
+                response = response.slice(0, 5)
+                // remove already added pills
+                response = response.filter((user) => {
+                    const foundPill = pills.find(pillObject =>
+                        pillObject.email == user.email
+                    )
+                    return !foundPill
+                })
+
+                setSearchResult(response.map((user: User) => {
+                    return {
+                        name: user.name,
+                        email: user.email
+                    }
+                }))
+            })
+            .catch((error) => console.log(error))
+    }
+        , [pills, token])
+
+    useEffect(() => {
+        if (!debouncedValue || debouncedValue.trim() === "") {
+            setSearchResult([])
+        }
+        else {
+            searchUsersApi(debouncedValue)
+        }
+    }, [debouncedValue, searchUsersApi]);
+
+    const handleDelete = (email: string) => {
+        setPills((prevState) => prevState.filter((pill) => pill.email != email))
+
+    }
+
+    const addUserToPills = (pill: Pill): void => {
+        const foundPill = pills.find(pillObject => pillObject.name == pill.name && pillObject.email == pill.email)
+        if (!foundPill) {
+            setNameSearchValue("")
+            setDebouncedValue("")
+            setSearchResult([])
+            setEmptyError(false)
+            setPills((prevState) => {
+                return [...prevState, pill];
+            });
+        }
+    }
+
+    const handleDebouncedInput = useDebouncedCallback(
+        (value) => {
+            setDebouncedValue(value);
+        },
+        500
+    );
+
+    const handleNameSearchChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        setShowSearchResults(true)
+        setNameSearchValue(e.target.value)
+        handleDebouncedInput(e.target.value)
+    }
+
+    const handleBlur = () => {
+        setTimeout ( () => {
+            setShowSearchResults(false)
+        }, 500)
+    }
+
+    return (
+        <div className={styles['acm-event-heads-container']}>
+            <div className={styles.pillInputDiv}>
+                <PillContainer handleDelete={handleDelete} pills={pills} />
+                <input
+                    type="text"
+                    onChange={handleNameSearchChange}
+                    value={nameSearchValue}
+                    className={styles.pillInput}
+                    placeholder={`Search ${teamTitle}*`}
+                    onBlur={handleBlur}
+                />
```

Comment:
Use `<TextFIeld/>` from `@mui/material` here, instead of plain `input`

This doesnt look right

![Screenshot 2023-11-20 at 12 22 55 PM](https://github.com/PEC-CSS/website/assets/50266759/ae396c47-71c2-4fbe-a1cb-d57d4f4c03d3)

Reply:
There is nothing we can do ://